test/UnitSession.cpp

```diff
@@ -232,6 +233,122 @@ UnitBase::TestResult UnitSession::testSlideShow()
     return TestResult::Ok;
 }
 
+/// Load a document on server and download it multiple to this client using 2 reused channels:
+/// - wsd-session for commands (websocket)
+/// - http-session to download it, as well as to load the favicon
+UnitBase::TestResult UnitSession::testSlideShowMultiDL()
+{
+    setTestname(__func__);
+    TST_LOG("Starting Test: " << testname);
+    const int dlCount = 10;
+    int dlIter = -1;
+    try
+    {
+        std::string documentPath, documentURL;
+        std::string response;
+        helpers::getDocumentPathAndURL("setclientpart.odp", documentPath, documentURL, testname);
+
+        // Reused http wsd session + socket-poll to send commands to server
+        std::shared_ptr<SocketPoll> wsdSocketPoll = std::make_shared<SocketPoll>(testname);
+        wsdSocketPoll->startThread();
```

Comment:
We can run the socket logic on the current thread if we set:
    bool runOnClientThread()
on the SocketPoll - which can make the tests simpler to follow.
Use 'SocketPoll::poll()' in a loop to wait for your condition then and avoid threads / races =)

Reply:
looking into it.. thx